frontend/packages/core-auth/src/components/AuthWrapper.tsx

```diff
@@ -83,29 +80,29 @@ const AuthWrapper: React.FC<AuthWrapperProps> = ({
       });
   }, [request?.codeVerifier, response, discovery]);
 
+  // Trigger onTokenChange callback when TokenResponse is received
   useEffect(() => {
-    if (!discovery) {
-      return;
-    }
-    if (tokenResponse?.shouldRefresh()) {
-      const refreshConfig = {
-        clientId,
-        scopes,
-        extraParams: {},
-      };
-      tokenResponse
-        ?.refreshAsync(refreshConfig, discovery)
-        .then((resp) => {
-          setTokenResponse(resp);
-        })
-        .catch((err) => {
-          setTokenResponse(undefined);
-        });
+    console.log('tokenResponse has changed...');
+
+    if (tokenResponse == null) return;
+
+    let token;
+    switch (injectToken) {
+      case 'access_token':
+        token = tokenResponse?.accessToken ?? '';
+        break;
+      case 'id_token':
+        token = tokenResponse?.idToken ?? '';
+        break;
+      default:
+        token = '';
     }
-  }, [tokenResponse?.shouldRefresh(), discovery]);
+    onTokenChange(token);
+  }, [JSON.stringify(tokenResponse)]);
```

Comment:
The functionality is nearly there, but for some reason that I don't understand, this `useEffect` hook is not getting triggered when the timeout triggers the token refresh.

The `setTokenResponse` call on line 160 doesn't cause a re-render; which is what I would [expect](https://stackoverflow.com/questions/46240647/react-how-to-force-a-function-component-to-render/53837442#53837442)...  I must be overlooking something

Reply:
`[JSON.stringify(tokenResponse)]` I don't think this will stringify that well. When using `JSON.stringify` it should be something json serialisable.

I'd replace this with `[tokenResponse?.accessToken, tokenResponse?.idToken]`